src/main/java/games/strategy/triplea/settings/SelectionComponentFactory.java

```diff
@@ -340,11 +344,26 @@ public void reset() {
     return selectFile(clientSetting, SwingComponents.FolderSelectionMode.DIRECTORIES);
   }
 
-  static Supplier<SelectionComponent> selectionBox(
+  static <T> Supplier<SelectionComponent> selectionBox(
       final ClientSetting clientSetting,
-      final List<String> availableOptions) {
+      final List<T> availableOptions,
+      final Function<T, ? extends Component> renderFunction) {
     return () -> new AlwaysValidInputSelectionComponent() {
-      final JComboBox<String> comboBox = new JComboBox<>(availableOptions.toArray(new String[availableOptions.size()]));
+      final JComboBox<T> comboBox = getCombobox();
+
+      private JComboBox<T> getCombobox() {
+        final JComboBox<T> comboBox = new JComboBox<>();
+        availableOptions.forEach(comboBox::addItem);
+        comboBox.setRenderer(new ListCellRenderer<T>() {
+
+          @Override
+          public Component getListCellRendererComponent(
+              JList<? extends T> list, T value, int index, boolean isSelected, boolean cellHasFocus) {
+            return renderFunction.apply(value);
```

Comment:
Note that this is not properly handling focus/selection rendering.  If you expand the combo box and try to move the selection around the item list, there is no visual feedback.  That's probably another reason to handle the renderer component internally to this method.

If you apply the suggestion above to convert `renderFunction` to `Function<T, String>`, then a `DefaultListCellRenderer` can be used as:

```java
comboBox.setRenderer(new DefaultListCellRenderer() {
  private static final long serialVersionUID = -1646566310567151596L;

  @Override
  @SuppressWarnings("unchecked")
  public Component getListCellRendererComponent(
      final JList<?> list,
      final Object value,
      final int index,
      final boolean isSelected,
      final boolean cellHasFocus) {
    return super.getListCellRendererComponent(
        list,
        renderFunction.apply((T) value),
        index,
        isSelected,
        cellHasFocus);
  }
});
```